addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoDeviceDefinitions.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.io.neeo.NeeoConstants;
+import org.openhab.io.neeo.internal.models.NeeoCapabilityType;
+import org.openhab.io.neeo.internal.models.NeeoDevice;
+import org.openhab.io.neeo.internal.models.NeeoDeviceChannel;
+import org.openhab.io.neeo.internal.models.NeeoDeviceType;
+import org.openhab.io.neeo.internal.serialization.NeeoCapabilityTypeSerializer;
+import org.openhab.io.neeo.internal.serialization.NeeoDeviceChannelSerializer;
+import org.openhab.io.neeo.internal.serialization.NeeoDeviceSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+
+/**
+ * This class manages device definitions. Device definitions map openHAB things/channels to NEEO device/capabilities.
+ * All device definition changes are saved to a JSON file in the user data folder.
+ *
+ * @author Tim Roberts - initial contribution
+ */
+public class NeeoDeviceDefinitions implements AutoCloseable {
```

Comment:
Why AutoCloseable, when close does not do anything

Reply:
That did have code at one point - but eliminated it.  I'll eliminate the autocloseable